client/sites-dashboard/components/wpcom-smp-commands.tsx

```diff
@@ -362,6 +372,26 @@ export const useCommandsArrayWpcom = ( {
 			},
 			icon: cacheIcon,
 		},
+		{
+			name: 'visitSite',
+			label: __( 'Visit site' ),
+			searchLabel: [
+				_x( 'visit site', 'Keyword for the Visit site dashboard command' ),
+				_x( 'see site', 'Keyword for the Visit site dashboard command' ),
+				_x( 'browse site', 'Keyword for the Visit site dashboard command' ),
+			].join( ' ' ),
+			context: [ '/:site' ],
+			callback: currentSite
+				? commandNavigation( currentSite.URL, { openInNewTab: true } )
+				: setStateCallback( 'visitSite', __( 'Select site to visit' ) ),
```

Comment:
This is a fancy way to handle it, although it may introduce some confusion to users?
Why this behaves differently from the rest of commands?

I would suggest using only the nested command way. The current site will be highlighted automatically, so it's not a huge effort for the user to pick that site and it's consistent with the rest of commands.